File: DSA-2019/week-02/day-1/station.py
from car import Car


class Statiom():
    def __init__(self, gas_amount):
        self.gas_amount = gas_amount

    def refill(self, car: Car):
        refill_amount = self.refill_amount(car)
        self.gas_amount -= refill_amount
        car.gas_amount += refill_amount

    def refill_amount(self, car: Car):
        if self.gas_amount >= ((car.capacity - car.gas_amount)):
            return car.capacity - car.gas_amount
        else:
            return self.gas_amount

File: DotNet-2018/week2/day2/week2-day2/week2-day2/Station.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace week2_day2
{
    class Station
    {
        public float gasAmount;

        public Station()
        {
            this.gasAmount = 500;
        }

        public void Refill(Car car)
        {
            this.gasAmount -= car.capacity;
            car.gasAmount += car.capacity;
        }

    }
}
